src/main/java/homer/model/airquality/AirQualityStateImpl.java

```diff
@@ -18,7 +18,11 @@ public final class AirQualityStateImpl implements AirQualityState {
      * @param pm25
      */
     public AirQualityStateImpl(final double co2, final double pm10, final double toxicGasPercentage, final double pm25) {
-        this.co2 = co2;
+        requireGreaterOrEqualThanZero(co2); 
+        requireGreaterOrEqualThanZero(pm10); 
+        requireGreaterOrEqualThanZero(toxicGasPercentage);
+        requireGreaterOrEqualThanZero(pm25); 
+        this.co2 =  co2;
```

Comment:
lgtm, if you want to save some lines you could use the setters in the constructor

Reply:
that's right